syz-runner/runner.go

```diff
@@ -89,17 +90,24 @@ func main() {
 	}
 
 	res := &rpctype.NextExchangeRes{}
-	if err := rn.vrf.Call("Verifier.NextExchange", &rpctype.NextExchangeArgs{Pool: rn.pool, VM: rn.vm}, res); err != nil {
-		log.Fatalf("failed to get initial program: %v", err)
+	for {
+		if err := rn.vrf.Call("Verifier.NextExchange", &rpctype.NextExchangeArgs{Pool: rn.pool, VM: rn.vm}, res); err != nil {
+			log.Fatalf("failed to get initial program: %v", err)
+		}
+
+		if res.ExecTaskID != rpctype.NoTask {
+			break
+		}
+		time.Sleep(1 * time.Second)
```

Comment:
Agree. Better approach is to block the request and wait till the task availability.
Don't want that lock to happen now because Server use mutex for all the requests (locking it now by pool0 NextExchange() will block the Connect & UpdateUnsupportedCalls from pool1).
My plan is to remove the verifier.srv.mu first and then rewrite runner.

Reply:
The Sleep is here.